### PR TITLE
Add nektar@master

### PR DESCRIPTION
**Pull request by oparry-ukaea**
Add nektar@master

### CODE REVIEW

File: packages/nektar/package.py [hunk 1]
```diff
@@ -16,35 +16,96 @@ class Nektar(CMakePackage):
...
+    version("master", branch="master")
     version("5.4.0", commit="002bf62648ec667e10524ceb8a98bb1c21804130", preferred=True)
     version("5.3.0", commit="f286f809cfeb26cb73828c90a689a048898971d2")
     version("5.2.0-2022-09-03", commit="2e0fb86da236e7e5a3590fcf5e0f608bd8490945")
 
-    patch("add_compflow_solver_lib_v5.2.0_2022-09-03.patch", when="@5.2.0-2022-09-03:5.3")
+    patch(
+        "add_compflow_solver_lib_v5.2.0_2022-09-03.patch", when="@5.2.0-2022-09-03:5.3"
+    )
     patch("add_compflow_solver_lib_v5.4.0.patch", when="@5.4.0:")
```
Comment (cmacmackin) [inline packages/nektar/package.py:27]: I think this line still has to be changed like in #33, doesn't it?

**Comment by oparry-ukaea** [comment]
My plan was to leave 
`depends_on("nektar@5.2.0-2022-09-03+compflow_solver", type="link")`
in NESO for now.

Then do a separate PR that switches to
`depends_on("nektar@master+compflow_solver", type="link")`
And adds all of the corresponding nektar build changes.
Does that make sense?

Or just put it all in this PR?

File: packages/nektar/package.py (hunk 1 above)
Comment (cmacmackin) [reply packages/nektar/package.py:27]: According to the manual:
> Finally, there are a few special string components that are considered “infinity versions”. They include develop, main, master, head, trunk, and stable. For example: 1.2 < develop. These are useful for specifying the most recent development version of a package (often a moving target like a git branch), without assigning a specific version number. Infinity versions are not automatically used when determining the latest version of a package unless explicitly required by another package or user.

So yes, you still need to get rid of the colon in line 24.

Comment (oparry-ukaea) [reply packages/nektar/package.py:27]: :+1: thanks, done

**Comment by oparry-ukaea** [comment]
> Sure, that's fine.

Was this in response to "does that make sense", or "shall I put it all in this PR"?

**Comment by cmacmackin** [comment]
To "does that make sense"

**Pull request merged**